### PR TITLE
Schedule disconnection timeouts

### PR DESCRIPTION
**Pull request by jchambers**
Rather than adding a second idle state handler to the pipeline, we can just schedule an immediate disconnection after some delay. This seems a whole lot simpler to me.

Also, renamed "shutdown" to "disconnect" in the context of `ApnsConnections` for clarity.

### CODE REVIEW

File: src/main/java/com/relayrides/pushy/apns/ApnsConnection.java [hunk 1]
```diff
@@ -326,6 +324,10 @@ public void channelInactive(final ChannelHandlerContext context) throws Exceptio
 					this.apnsConnection.listener.handleConnectionClosure(this.apnsConnection);
 				}
 			}
+
+			if (this.apnsConnection.gracefulShutdownTimeoutFuture != null) {
```
Comment (AndreaLev) [inline src/main/java/com/relayrides/pushy/apns/ApnsConnection.java]: Very minor: why does this still have `shutdown` when elsewhere that is renamed to `disconnect`?

**Comment by AndreaLev** [comment]
Overall I think this is a good approach :+1:

File: src/main/java/com/relayrides/pushy/apns/ApnsConnection.java (hunk 1 above)
Comment (jchambers) [inline src/main/java/com/relayrides/pushy/apns/ApnsConnection.java]: Oops—just a goof on my part. Fixed in c4bab82.

**Comment by jchambers** [comment]
Going to merge this even though Travis CI is reporting errors; I suspect it might be related to their [ongoing build environment update](http://www.traviscistatus.com/incidents/yk24wrkb4082).

**Pull request merged**